internal/testprocessor/testscript.go

```diff
@@ -0,0 +1,128 @@
+package testprocessor
+
+import (
+	"encoding/csv"
+	"fmt"
+	"os"
+	"time"
+
+	zlog "github.com/rs/zerolog/log"
+)
+
+type Mode string
+
+const (
+	SerialMode              Mode = "serial"
+	SerialWithoutDelaysMode Mode = "serial-without-delays"
+	ParallelMode            Mode = "parallel"
+)
+
+type Config struct {
+	Mode         Mode
+	RunsDataPath string
+	OutputPath   string
+	DefaultQuery *string
+	Percentiles  []int
+}
+
+type PlaygroundClient interface {
+	PostRuns(database string, version string, query string) (time.Duration, error)
+}
+
+type Processor struct {
+	PlaygroundClient PlaygroundClient
+	Config           *Config
+}
+
+func New(playgroundClient PlaygroundClient, config *Config) *Processor {
+	return &Processor{
+		PlaygroundClient: playgroundClient,
+		Config:           config,
+	}
+}
+
+func (p *Processor) Process() {
+	runs, err := loadRuns(p.Config.RunsDataPath)
+	if err != nil {
+		zlog.Fatal().Err(err).Msg("runs data cannot be loaded")
+	}
+
+	if p.Config.DefaultQuery != nil {
+		for _, run := range runs.Runs {
+			run.Query = *p.Config.DefaultQuery
+		}
+	}
+
+	switch p.Config.Mode {
+	case SerialMode:
+		p.processSerialMode(runs)
```

Comment:
Interfaces seem to be more common solution to that.

Think about something like that:
```go
type Processor interface {
  Mode() Mode
  Process(runs Runs) returnTypeIfNeeded
}
```

Although the interface specifies how to use them, init implementation is different for different processors.